2022/02/index.js

```diff
@@ -2,13 +2,93 @@ const { promises: fs } = require('fs');
 
 async function getInput() {
   const input = (await fs.readFile(`${__dirname}/input.txt`))
-    .toString();
+    .toString()
+    .trim();
+
+  return input
+    .split('\n')
+    .map(round => round.split(' '));
 }
 
 async function p1() {
+  const selectionScore = {
+    'X': 1, // rock
+    'Y': 2, // paper
+    'Z': 3, // scissors
+  };
+
+  const opponentToYourMove = {
+    'A': 'X', // rock
+    'B': 'Y', // paper
+    'C': 'Z', // scissors
+  }
+
+  const input = await getInput();
+  return input.reduce(
+    (acc, [opponentMove, myMove]) => {
+      let totalScore = acc + selectionScore[myMove];
+
+      if(opponentToYourMove[opponentMove] === myMove) {
+        return totalScore + 3;
+      }
+
+      if(
```

Comment:
This looks like it should be else if, but I don't think that makes any difference because of the return in the if above

Reply:
Yeah I like early returns if possible.